### ISSUE TITLE
Consider updating package.json exports

### ISSUE THREAD

**Issue by izikaj** [mondaycom/monday-ui-react-core#1423]
Currently, the package has broken typing exports for TS.

## Steps to reproduce: 
- use Node 18
- `npm create vite@latest` & choose React + TS
- `npm add -S monday-ui-react-core`
- add something like this to the `src/App.tsx`
  near imports:
  ``` TypeScript
  import "monday-ui-react-core/tokens";
  import { Subitems } from "monday-ui-react-core/icons";
  import { IconButton } from "monday-ui-react-core";
  ```
  in the JSX-template
  ```JSX
  <IconButton icon={Subitems} />
  ```
 
As a result - all works, but you will receive 2 issues with typings:
For `monday-ui-react-core`: 
```
Could not find a declaration file for module 'monday-ui-react-core'. '/www/dummy/node_modules/monday-ui-react-core/dist/esm/src/index.js' implicitly has an 'any' type.
  There are types at '/www/dummy/node_modules/monday-ui-react-core/dist/types.d.ts', but this result could not be resolved when respecting package.json "exports". The 'monday-ui-react-core' library may need to update its package.json or typings.ts(7016)
Could not find a declaration file for module 'monday-ui-react-core'. '/www/dummy/node_modules/monday-ui-react-core/dist/esm/src/index.js' implicitly has an 'any' type.
```
And for `monday-ui-react-core/icons`:
```
Could not find a declaration file for module 'monday-ui-react-core/icons'. '/www/dummy/node_modules/monday-ui-react-core/dist/esm/src/components/Icon/Icons/icons.js' implicitly has an 'any' type.
  There are types at '/www/dummy/node_modules/monday-ui-react-core/dist/esm/components/Icon/Icons/index.d.ts', but this result could not be resolved when respecting package.json "exports". The 'monday-ui-react-core' library may need to update its package.json or typings.ts(7016)
```

## Method to solve the problem:

All the problems are easily solved by a patch of exports part of the package.json:
```
diff --git a/node_modules/monday-ui-react-core/package.json b/node_modules/monday-ui-react-core/package.json
index 0511b71..1309b93 100644
--- a/node_modules/monday-ui-react-core/package.json
+++ b/node_modules/monday-ui-react-core/package.json
@@ -8,10 +8,14 @@
     "./package.json": "./package.json",
     ".": {
       "import": "./dist/esm/src/index.js",
-      "default": "./dist/main.js"
+      "default": "./dist/main.js",
+      "types": "./dist/types.d.ts"
     },
     "./dist/": "./dist/",
-    "./icons": "./dist/esm/src/components/Icon/Icons/icons.js",
+    "./icons": {
+      "default": "./dist/esm/src/components/Icon/Icons/icons.js",
+      "types": "./dist/types/components/Icon/Icons/index.d.ts"
+    },
     "./interactionsTests": "./dist/esm/src/tests/interactionsTests.js",
     "./testIds": "./dist/esm/src/tests/testIds.js",
     "./storybookComponents": "./dist/storybook/index.js",
```

I'm looking forward to not using such patches, but receiving proper package.json instead.

**Comment by shlomitc** [comment]
Hey @izikaj - can you please try using this pre-release version and tell me if it works?
`npm i monday-ui-react-core@2.17.2-export-types-for-next-950406bb23.0`

**Comment by izikaj** [comment]
Hi @shlomitc, yes it works perfectly 👍

**Comment by shlomitc** [comment]
Merged and released in 2.17.2

**Comment by miliklukasz** [comment]
I am trying to test component using Vibe design system. During tests I am getting an error: 
` Cannot find module 'monday-ui-react-core/icons'` 
The app builds normally and development and production environment works fine. Only Jest cannot import module. Is there a way to fix it? Above solution doesn't work.
